### PR TITLE
MM-11693 Allow connections to /plugins for interactive message buttons.

### PR DESCRIPTION
**Pull request by crspeller**

### CODE REVIEW

**Comment by crspeller** [comment]
@lieut-data Yes, your right. The one case this doesn't cover is referring too the local machine by host name that resolves to an internal address. I didn't bother with adding the SiteURL because I assumed it would be an external address and that would work without modification.

**Comment by cpanato** [comment]
change the base to `release-5.3`

**Comment by crspeller** [comment]
@lieut-data Ready for re-review.

File: app/post.go [hunk 1]
```diff
@@ -882,7 +882,18 @@ func (a *App) DoPostAction(postId, actionId, userId, selectedOption string) *mod
 	req, _ := http.NewRequest("POST", action.Integration.URL, strings.NewReader(request.ToJson()))
 	req.Header.Set("Content-Type", "application/json")
 	req.Header.Set("Accept", "application/json")
-	resp, err := a.HTTPClient(false).Do(req)
+
+	// Allow access to plugin routes for action buttons
+	var httpClient *http.Client
+	url, _ := url.Parse(action.Integration.URL)
+	siteURL, _ := url.Parse(*a.Config().ServiceSettings.SiteURL)
+	if (url.Hostname() == "localhost" || url.Hostname() == "127.0.0.1" || url.Hostname() == siteURL.Hostname()) && strings.HasPrefix(url.Path, "/plugins") {
```
[the remark this answers is not in the document]
Comment (crspeller) [reply app/post.go]: Fixed.

File: app/post.go [hunk 2]
```diff
@@ -887,7 +888,8 @@ func (a *App) DoPostAction(postId, actionId, userId, selectedOption string) *mod
 	var httpClient *http.Client
 	url, _ := url.Parse(action.Integration.URL)
 	siteURL, _ := url.Parse(*a.Config().ServiceSettings.SiteURL)
-	if (url.Hostname() == "localhost" || url.Hostname() == "127.0.0.1" || url.Hostname() == siteURL.Hostname()) && strings.HasPrefix(url.Path, "/plugins") {
+	subpath, _ := utils.GetSubpathFromConfig(a.Config())
+	if (url.Hostname() == "localhost" || url.Hostname() == "127.0.0.1" || url.Hostname() == siteURL.Hostname()) && strings.HasPrefix(url.Path, path.Join(subpath, "plugins")) {
```
Comment (lieut-data) [inline app/post.go]: If `GetSubpathFromConfig()` fails, `subpath` will be `""`, and the `HasPrefix` check will fail. That being said, I think this is an edge case that indicates other things as broken, so your simplification of this check makes sense :)

Comment (crspeller) [reply app/post.go]: Yup. I didn't want that failing to cause the other cases that don't require it to fail.

**Pull request merged**